docs/docs/creating-a-generic-plugin.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Creating a Generic plugin
+---
+
+This section is aimed at explaining the structure of a Gatsby plugin and the files that are needed to create it.
+
+As seen in the [what is a plugin doc](/docs/what-is-a-plugin/), a plugin is a piece of software that acts as an add-on and gives your site additional functionality.
+
+They contain a file, usually in the project root, called `package.json` - this file holds various metadata relevant to the project. This file is used to give information to npm that allows it to identify the project as well as handle the project's dependencies.
+
+## Initializing your plugin project
+
+The process of initializing a `package.json` in a project is done by running:
+
+```shell
+  npm init
+```
+
+After running the above command a series of options are listed in the command line interface (CLI). Those you select are stored in your `package.json` which contains some of the [files Gatsby looks for in a Plugin](/docs/docs/files-gatsby-looks-for-in-a-plugin)
+
+## What happens in a Generic Plugin?
+
+In a generic plugin the `gatsby-node.js` file enables the use of [gatsby node APIs](/docs/node-apis/) such as `createPage`, `createResolvers`, and `sourceNodes`. A Node is the smallest unit of data in Gatsby. Nodes are created using the [createNode](/docs/actions/#createNode) action
```

Comment:
```suggestion
In a generic plugin the `gatsby-node.js` file enables the use of [gatsby node APIs](/docs/node-apis/). These APIs, such as `createPage`, `createResolvers`, and `sourceNodes`, manipulate the Node(s) in a Gatsby site. A Node is the smallest unit of data in Gatsby. You can create a Nodes using the [createNode](/docs/actions/#createNode) action.
```
Can we expand on what a Node is? Or link to something that does? I think that's relevant for why someone would want to manipulate one.

Reply:
Gotcha! I just added a link to a better explanation